OP-1169 | DELETE last movement (only) in Pharmaceutical Stock Ward (#240)
informatici/openhospital-doc: doc_user/UserManual.adoc

Before pressing the *[.underline]##O##K* button, insert as many Medicals as needed; it will show a corresponding number of movements in the *Outcomes* tab of the *Movement panel* in the *_Ward Pharmacy_* window. These movements are also visible in the *Incomings* tab of the *Movement panel* in the *_Ward Pharmacy_* window of the receiving ward, in case of discharging to another ward.

> ===== 4.3.2.5 Delete the last stock movement in ward ([.underline]##D##elete)

Reviewer: ```suggestion
===== 4.3.2.5 Delete the Last Stock Movement in a Ward ([.underline]##D##elete)
```
Match the other subsection title format

> In a pharmaceutical stock ward, it is possible to discharge medicals from *patient*, *internal use* or *another ward*. If a mistake is made during the discharge, it is possible to delete it in order to correct the stock of the ward and resume the action.

Reviewer 2: ```suggestion
In a pharmaceutical stock ward, it is possible to discharge medicals to *patient*, *internal use* or *another ward*. If a mistake is made during the discharge, it is possible to delete it in order to correct the stock of the ward and resume the action.
```

> In pharmaceutical stock ward, we have possibility to discharge medical from *patient*, *internal use* or *another ward*. if you make a mistake during the discharge you have the possibility to delete it in order to correct the stock of the ward and resume the action if possible.

Reviewer: ```suggestion
In a pharmaceutical stock ward, it is possible to discharge medicals from *patient*, *internal use* or *another ward*. If a mistake is made during the discharge, it is possible to delete it in order to correct the stock of the ward and resume the action.
```

> The Administrator can enable or disable the *[.underline]##D##elete* button that allows the deletion of the last movement.  This aids in quickly undoing mistakes.

Reviewer 2: ```suggestion
The Administrator can enable or disable the *[.underline]##D##elete* button that allows the deletion of the last movement. This aids in quickly undoing mistakes.
```

> Nevertheless, the admin can enable/disable the *[.underline]##D##elete* button that allows the enabled user do delete only the last movement, and helps quickly undoing mistakes.

Reviewer: ```suggestion
The Administrator can enable or disable the *[.underline]##D##elete* button that allows the deletion of the last movement.  This aids in quickly undoing mistakes.
```

> To delete the last movement, select first the row that you want to delete and press the Delete button in the Stock Movement Browser window. Only one row is allowed to delete at a time.

Reviewer: ```suggestion
To delete the last movement, first select the row to delete and then press the Delete button in the Stock Movement Browser window. Only one row can be deleted at a time.
```

> If it is not the last movement the following Error message is shown:

Reviewer: ```suggestion
If the selection is not the last movement the following error message dialog is shown:
```

image:image250.png[Not last movement error]

> But if the row selected is the last movement this *_Question_* message is presented:

Reviewer: ```suggestion
If the row selected is the last movement then this *_Question_* message is presented:
```

image:image248.png[Question]

> Click *[.underline]##Y##es* to confirm the deletion of the last movement, otherwise click *[.underline]##N##o* to cancel the operation. After clicking the *[.underline]##Y##es* button, the system checks if the product involved in the selected movement has not been used itself in the *_Pharmaceuticals Stock [.underline]##W##ard_* if the last movement is discharge from ward to ward.

Reviewer: ```suggestion
Click *[.underline]##Y##es* to confirm the deletion of the last movement, otherwise click *[.underline]##N##o* to cancel the operation. After clicking the *[.underline]##Y##es* button, the system checks if to see if the product involved in the selected movement and has not been used in the *_Pharmaceuticals Stock [.underline]##W##ard_*.   If not then the last movement is discharged from ward to ward.
```

> Click *[.underline]##Y##es* to confirm the deletion of the last movement, otherwise click *[.underline]##N##o* to cancel the operation. After clicking the *[.underline]##Y##es* button, the system checks if to see if the product involved in the selected movement and has not been used in the *_Pharmaceuticals Stock [.underline]##W##ard_*.   If not then the last movement is discharged from ward to ward.

Reviewer 2: ```suggestion
Click *[.underline]##Y##es* to confirm the deletion of the last movement, otherwise click *[.underline]##N##o* to cancel the operation. After clicking the *[.underline]##Y##es* button, if the movement discharges to another ward, the system checks if the product involved in the selected movement has not been used in the destination ward. If not, then the movement is deleted, otherwise the deletion is denied.
```

> If the product has been used in a ward, then the *_Error_* message is shown:

Reviewer 2: ```suggestion
If the product has been used in the destination ward, then the *_Error_* message is shown:
```

> If the product has been already used in a Ward, then the *_Error_* message is shown.

Reviewer: ```suggestion
If the product has been used in a ward, then the *_Error_* message is shown:
```

image:image249.png[Error]

Otherwise the selected movement is deleted and the *_Success_* window is shown:

image:image251.png[success]

> NOTE: This button can be enabled/disabled to allow/not allow movements deletion. Ask the Administrator or check the Administrator's Guide for more information.

Reviewer: ```suggestion
NOTE: This button can be enabled or disabled to control movements deletions. Ask the Administrator or check the Administrator's Guide for more information.
```
